### PR TITLE
refactor code, add support for configuring bigqueue

### PR DESCRIPTION
**Pull request by mangalaman93**
This patch adds support for setting custom Arena size through function options. The pattern used is described here https://commandcenter.blogspot.com/2014/01/self-referential-functions-and-design.html. See issue #5 for more details on this patch.

### CODE REVIEW

File: write.go [hunk 1]
```diff
@@ -36,45 +18,58 @@ func (bq *BigQueue) Enqueue(message []byte) error {
...
 
 	return nil
 }
 
+func (bq *BigQueue) writeLength(aid, offset int, length uint64) (int, int, error) {
+	// check if length can be fit into same arena, if not, get new arena
+	if cInt64Size+offset <= bq.arenaSize {
+		bq.arenaList[aid].WriteUint64(offset, length)
+	} else {
+		aid++
```
Comment (ashish-goswami) [inline write.go]: this line should be removed

Comment (mangalaman93) [reply write.go]: done

File: bigqueue_test.go [hunk 2]
```diff
@@ -203,55 +204,190 @@ func TestEnqueueLargeNumberOfMessages(t *testing.T) {
...
+	if err != nil {
+		t.Errorf("unable to dequeue :: %v", err)
+	}
+
+	if !bytes.Equal(msg, poppedMsg2) {
+		t.Errorf("unequal length, eq: %s, dq: %s", string(msg), string(poppedMsg2))
+	}
+}
+
+func TestArenaSize2(t *testing.T) {
```
Comment (ashish-goswami) [inline bigqueue_test.go]: TestArenaSize2 and TestArenaSize3 are similar except arena size argument to NewBigQueue. Check if you can merge both tests using subtests.

Comment (mangalaman93) [reply bigqueue_test.go]: leave it for now, I am planning to add more test later. Need another approval

**Pull request merged**